### PR TITLE
Feature #50

### LINKED ISSUE #50: Student registration pop-up (feature, user story #8)
Added student registration popup, added registration form and fild validation.

Image

[Video](https://www.loom.com/share/7b84dcedaede43319f1d21d851804fc3?sid=5b03d757-80d9-4579-a1eb-2a6608989cf0)

### PR DESCRIPTION
**Pull request by olehpona**

### CODE REVIEW

**Review by tufned — APPROVED** [review]

**Review by MrAK-dev — COMMENTED** [review]
Functionality and code looks fine, But i saw  few issues,
1. When i opened popup, i already see validation errors, but I don`t touch this fields yet.
image
2. In task in mockup first active input is first name:
image
In yours solution first active input is email - 
image

**Review by SeredenkoDS — APPROVED** [review]
Clicked the "Become a student" button, the form appears, the first field is active and there are no red errors (must have been fixed since the previous review). Looks good, approving.

File: src/containers/guest-home-page/student-registration-form/StudentRegistrationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import { useTranslation } from 'react-i18next'
+import useInputVisibility from '~/hooks/use-input-visibility'
+import { useSelector } from 'react-redux'
```
Comment (vlad-khrychov) [inline src/containers/guest-home-page/student-registration-form/StudentRegistrationForm.tsx]: Look at custom `use-redux` hook for fixing warning

File: src/containers/guest-home-page/student-registration-dialog/StudentRegistrationDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  )
+
+  const onCloseAction = useCallback(() => {
+    Object.values(data).some((value) => Boolean(value))
+      ? setShowConfirm(true)
+      : closeModal()
+  }, [data, closeModal])
+
+  useEffect(() => setCloseCallback(() => onCloseAction), [onCloseAction])
```
Comment (vlad-khrychov) [inline src/containers/guest-home-page/student-registration-dialog/StudentRegistrationDialog.tsx]: Could we add `setCloseCallback` to dep. array? If it breaks the logic, add a comment to skip the line in ES lint.

**Review by MrAK-dev — APPROVED** [review]

File: src/containers/guest-home-page/student-registration-form/StudentRegistrationForm.tsx (hunk 1 above)
Comment (olehpona) [reply src/containers/guest-home-page/student-registration-form/StudentRegistrationForm.tsx]: changed code to use typed useAppSelector

File: src/containers/guest-home-page/student-registration-dialog/StudentRegistrationDialog.tsx (hunk 2 above)
Comment (olehpona) [reply src/containers/guest-home-page/student-registration-dialog/StudentRegistrationDialog.tsx]: Added setClsoeCallback to deps

**Review by HalynaProlubshchykova — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]
Well done!

**Pull request merged**